compiler/rustc_passes/src/check_attr.rs

```diff
@@ -1373,6 +1373,28 @@ impl<'tcx> CheckAttrVisitor<'tcx> {
         }
     }
 
+    /// Checks if `#[may_dangle]` is applied to a lifetime or type generic parameter in `Drop` impl.
+    fn check_may_dangle(&self, hir_id: HirId, attr: &Attribute, target: Target) {
+        if matches!(target, Target::GenericParam(_))
```

Comment:
this first `matches` is redundant if we are just checking the node one line below (1379)

Reply:
Fair, I removed this check.